### DISCUSSION TITLE
SSTP IPv6 support

### DISCUSSION THREAD

**Discussion by AGluk** [SoftEtherVPN/SoftEtherVPN#2070] [Q&A]
Hello! Please help me to understand, is it possible to setup SoftEther for IPv6 support for SSTP clients?
I've tried to create TUN device and setting IPv6 address for it. Setting RADVd service, and other recommendations.
Nothing helped. It looks like I don't fully understand how it works :( ask for your help...
I've tried to connect from iPhone over SSTP Connect App and from macOS over sstpc tool. No one get IPv6 addres, only IPv4. So I can't reach to IPv6 internet servers over SSTP connection, only IPv4.
Thank you!

**Comment by Evengard** [comment]
I'm actually using SSTP with IPv6 successfully (well, I actually coded in the support, so yeah, I'm using it). The fact is that this should usually be manually enabled **on the client side** (if I remember well - even on Windows which is the native system for SSTP, other third party apps I was using too required that). Otherwise it should just work with standard protocols.

Personally, I use it with DNSMasq and giving out IPv6s via DHCP protocol instead of using classic autoconfiguration - seemed to work better with the VPN setups. I did had some success with autoconfiguration though too.

Note that you may need to enable IPv6 forwarding for it to work in sysctl.conf:
`net.ipv6.conf.all.forwarding=1`

Also, I've actually bridged my VPN hub to a virtual tunnel device (created through SoftEther itself) to configure it, so that may be important too.

**Comment by AGluk** [comment]
Thank you very much for the detailed response. For now, I’ve switched to Xray, where IPv6 support was enabled right away. However, if needed, I will continue to work on SSTP later.
